test/test-fs.c

```diff
@@ -3121,3 +3121,73 @@ TEST_IMPL(fs_exclusive_sharing_mode) {
   return 0;
 }
 #endif
+
+
+#if defined(__APPLE__)
+static void pipe_write_cb(uv_write_t* req, int status) {
+  ASSERT(status == 0);
+  write_cb_count++;
+  uv_close((uv_handle_t*)req->handle, (uv_close_cb)free);
+}
+#endif
+
+
+TEST_IMPL(big_write) {
+  const unsigned nzeros =
+#ifdef _WIN32
+    (1u << 30u) + 1u;
+#else
+    (1u << 31u) + 1u;
+#endif
+  uv_file file;
+  int r;
+  void *zeros;
+  const char dev_null[] =
+#ifdef _WIN32
+      "NUL";
+#else
+      "/dev/null";
+#endif
+#if defined(__APPLE__)
+  uv_write_t pipe_write;
+  uv_pipe_t *p;
+#endif
+
+  /* Setup. */
+  loop = uv_default_loop();
+
+  r = uv_fs_open(loop,
+                 &open_req1,
+                 dev_null,
+                 O_WRONLY,
+                 0,
+                 NULL);
+  ASSERT(r > 0);
+  ASSERT(open_req1.result >= 0);
+  file = (uv_file)open_req1.result;
+  uv_fs_req_cleanup(&open_req1);
+
+  zeros = zalloc_2gb();
+  ASSERT(zeros);
+  iov = uv_buf_init(zeros, nzeros);
+
+#if defined(__APPLE__) /* usually not OK to treat a fd as a pipe */
+  p = (uv_pipe_t*)malloc(sizeof(uv_pipe_t));
```

Comment:
Superfluous cast and use `sizeof(*p)` (but why heap-allocate it when you can just stack-allocate it?)